src/extension/layer-filters/src/main/java/au/org/emii/geoserver/extensions/filters/LayerFiltersService.java

```diff
@@ -148,6 +153,11 @@ public void uniqueValues(HttpServletRequest request, HttpServletResponse respons
         String layer = request.getParameter("layer");
         String propertyName = request.getParameter("propertyName");
 
+        if(workspace==null) {
```

Comment:
Yeah I was in two minds about that, feels slightly more readable with two calls I think, but also a bit wasteful. Have made change you suggest.